city_scrapers/spiders/det_employment_solutions_corp.py

```diff
@@ -0,0 +1,183 @@
+            except ValueError:
+                date_str = date_str.replace(" ", "", 1)
+            time_str = time_tuples[i]
+            start_time = time_str[0]
+            end_time = time_str[1]
+            self.meeting_dates.append({
+                'start': self._parse_datetime(start_time, date_str, year_str),
+                'end': self._parse_datetime(end_time, date_str, year_str)
+            })
+
+    def _parse_documents(self, response):
+        """Parse agenda and minutes page"""
+        link_map = self._parse_link_map(response)
+        title_map = self._parse_title_map(response)
+        for date in self.meeting_dates:
+            title_in = self._parse_title(date, title_map)
+            meeting = Meeting(
+                title=title_in,
+                description="",
+                classification=self._parse_classification(title_in),
+                start=date['start'],
+                end=date['end'],
+                all_day=False,
+                time_notes="",
+                location=self.location,
+                links=link_map[(date['start'].month, date['start'].year)],
+                source=self.start_urls[0],
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_title(self, date, title_map):
+        """Generate title from title map or return generic title"""
+        if title_map[(date['start'].month, date['start'].year)]:
+            return title_map[(date['start'].month, date['start'].year)]
+        return "Detroit Employment Solutions Corporation"
+
+    def _parse_classification(self, title):
+        """Parse or generate classification from allowed options."""
+        if "Board" in title:
+            return BOARD
+
+        if "Committee" in title:
+            return COMMITTEE
+
+        return NOT_CLASSIFIED
+
+    def _parse_datetime(self, time_str, date_str, year_str):
+        """Parse start datetime as a naive datetime object."""
+        return datetime.strptime(
+            "{} {} {}".format(date_str, year_str, time_str), "%B %d %Y %I:%M %p"
+        )
+
+    def _parse_link_map(self, response):
+        """Parse or generate links. Returns a dictionary of month, year tuples and link lists"""
+        link_map = defaultdict(list)
+
+        title_divs = response.css('div[class="meeting-min_inner-wrapper"]')
+        for div in title_divs:
+            date = div.css("p::text").extract_first().strip()
+            date = date.split('/')
+            link_start = datetime(int(date[2]), int(date[0]), int(date[1]))
+            link = div.css("a").xpath('@href').get()
+            print(link)
+            link_map[(link_start.month, link_start.year)].append({"title": "Minutes", "href": link})
+
+        return link_map
+
+    def _parse_title_map(self, response):
+        """parse or generate titles Returns a dictionary of month, year tuples and title lists"""
```

Comment:
Related to the above comment, it looks like there are multiple types of meetings on the same day

Reply:
Thanks for the feedback! I will work on these issues and resubmit (along with more robust tests to detect them).

Reply:
It also looks like there are some meetings on the minutes page that are not public. Should those be included as well?

Reply:
Any in particular that don't seem to be public? I've only skimmed it but I haven't seen any. In general, if we can safely ignore ones that aren't public without accidentally ignoring public ones we should give it a try, but even the conference calls still qualify as public meetings

Reply:
Specifically, the Mayor’s Workforce Development Board meetings don't show up in the public meeting schedule pdf but minutes are listed on the minutes page. I have fixed the link parser to look up by title as well, but it seems like I will then need to find a way to get the titles out of the pdf itself, since it seems difficult to assign the right titles to two meetings that occur on the same day using the minutes page.